dartagnan/src/main/java/com/dat3m/dartagnan/program/processing/ThreadCreation.java

```diff
@@ -128,8 +128,8 @@ public void run(Program program) {
                         comAddress.setInitialValue(0, expressions.makeZero(archType));
 
                         final List<Event> replacement = eventSequence(
-                                createEvent,
                                 newReleaseStore(comAddress, expressions.makeTrue()),
+                                createEvent,
```

Comment:
Why do we need to move the event down?

Reply:
Because under unfair scheduling, the thread can get spawned without the release store signaling the start.
This causes immediate non-termination.